src/test/java/com/snowflake/kafka/connector/SnowflakeSinkTaskForStreamingIT.java

```diff
@@ -41,109 +41,114 @@
  * com.snowflake.kafka.connector.internal.streaming.SnowflakeSinkServiceV2}
  */
 public class SnowflakeSinkTaskForStreamingIT {
-
+  private int taskId;
+  private int partitionCount;
   private String topicName;
-  private static int partition = 0;
-  private TopicPartition topicPartition;
   private Map<String, String> config;
-
-  @Mock Logger logger = Mockito.mock(Logger.class);
-
-  @InjectMocks @Spy
-  private LoggerHandler loggerHandler = Mockito.spy(new LoggerHandler(this.getClass().getName()));
-
-  @InjectMocks private SnowflakeSinkTask sinkTask1 = new SnowflakeSinkTask();
-
+  private List<TopicPartition> topicPartitions;
+  private SnowflakeSinkTask sinkTask;
+  private InMemorySinkTaskContext sinkTaskContext;
+  private List<SinkRecord> records;
+  private Map<TopicPartition, OffsetAndMetadata> offsetMap;
+
+  // sets up default objects required for sunny day tests
+  // NOTE: topicPartitions, records and corresponding offsetmap have a count of one each
   @Before
-  public void setup() {
-    topicName = TestUtils.randomTableName();
-    topicPartition = new TopicPartition(topicName, partition);
-
-    // config
-    this.config = TestUtils.getConfForStreaming();
-    this.config.put(BUFFER_COUNT_RECORDS, "1"); // override
-    this.config.put(INGESTION_METHOD_OPT, IngestionMethodConfig.SNOWPIPE_STREAMING.toString());
-    SnowflakeSinkConnectorConfig.setDefaultValues(this.config);
+  public void setup() throws Exception {
+    this.taskId = 0;
+    this.partitionCount = 1;
+    this.topicName = "topicName";
+    this.config = this.getConfig(this.taskId);
+    this.topicPartitions = getTopicPartitions(this.topicName, this.partitionCount);
+    this.sinkTask = new SnowflakeSinkTask();
+    this.sinkTaskContext =
+        new InMemorySinkTaskContext(Collections.singleton(this.topicPartitions.get(0)));
+    this.records = TestUtils.createJsonStringSinkRecords(0, 1, this.topicName, 0);
+    this.offsetMap = new HashMap<>();
+    this.offsetMap.put(this.topicPartitions.get(0), new OffsetAndMetadata(10000));
 
     IngestSdkProvider.getStreamingClientManager()
         .createAllStreamingClients(this.config, "testkcid", 2, 1);
+    assert IngestSdkProvider.getStreamingClientManager().getClientCount() == 2;
```

Comment:
this is fine for test but since partition count is only 1, should we also keep task count to 1? Just more relatable IMO.

Reply:
yeah entirely right, not sure how i missed this lol